### PR TITLE
Add Tests

### PR DESCRIPTION
**Pull request by Ewoskyx**
### Tests

In this update we did ;

- Add unit tests for operate.js and calculate.js functions
- Add unit tests for app components by using Jest - React Test

### CODE REVIEW

File: src/__tests__/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+it('matches snapshot 2 of Calculator component', () => {
+  const clickHandler = () => true;
+  const buttonName = '9';
+  const tree = renderer.create(<Calculator
+    buttonName={buttonName}
+    clickHandler={clickHandler}
+  />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Lameck1) [inline src/__tests__/Calculator.test.js]: - [OPTIONAL] I believe one snapshot test for the whole component would have been enough. Well done for experimenting 👏🏼, but I urge you to consider writing one snapshot test per component. There is never a need for separate snapshot tests for a single component.

File: src/__tests__/Calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+it('renders the props of component correctly', () => {
+  const clickHandler = () => true;
+  const buttonName = 'AC';
+  const { getByTestId } = render(<Calculator
+    buttonName={buttonName}
+    clickHandler={clickHandler}
+  />);
+  expect(getByTestId('calc')).toHaveTextContent('AC');
+});
```
Comment (Lameck1) [inline src/__tests__/Calculator.test.js:28]: - Tests added for the Calculator component are all acceptable. However, tests that assert that a user can interact with the calculator to perform a calculation are not visible anywhere. We need to prove that the user can click a combination of keys including the `=` sign and assert that the correct result is displayed. Please add this kind of test.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @Ewoskyx & @Mosams, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/Calculator.test.js (hunk 1 above)
Comment (Ewoskyx) [reply src/__tests__/Calculator.test.js]: Thank you for the suggestion. We will remove the obsolete snapshot tests. :)

File: src/__tests__/Calculator.test.js (hunk 2 above)
Comment (Ewoskyx) [reply src/__tests__/Calculator.test.js:28]: Thank you, this makes sense. We will apply this :)

**Review by oloomoses — APPROVED** [review]
Hi @Ewoskyx, 

Great job working on the suggested changes.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**